media/windows/ECC-MQTT-IoT/ECC_MQTT_IoT_SQLite.py

```diff
@@ -101,16 +124,149 @@
     print(F"ECC MQTT IoT Listener application, version {eccmqtt_iot_version}, {eccmqtt_iot_date}...")
     sys.exit(0)
 
+
+class GZipRotator(logging.handlers.RotatingFileHandler):
+
+    def doRollover(self):
+
+        global done_flag
+
+        # Let the default log rollover routine do its thing first...
+        super().doRollover()
+
+        # Construct a file pattern to match all current gzip'd logs
+        base_file_pattern = self.baseFilename + ".[0-9].gz"
+        # Get a list of the current .gzip'd log files to determine the next extension to use
+        arch_files = glob.iglob(base_file_pattern)
+        # Loop through the archived log files to find the last file number in use
+        lastgz_file_num = '0'  # default value for first iteration (no current gzips)
+        for archFile in arch_files:
+            fn = archFile.split('.')
+            # The gzip'd file "number" is in array element 2
+            lastgz_file_num = fn[2]
+        lastgz_file_int = int(lastgz_file_num)
+        # Construct a gzip file name for the destination by incrementing the last file number
+        # used.  If more than (backupCount) files are already zipped, delete the oldest and rename
+        # the remaining to free up the last number for re-use.  This ensures that the newest gzip
+        # will always be the highest number.
+        nextgz_file_int = lastgz_file_int + 1
+        if nextgz_file_int > self.backupCount:
+            nextgz_file_int = self.backupCount
+            # Since we already have the maximum gzip'd logs, delete the oldest, then rename the
+            # remaining to free up the last slot.
+            oldest_gz_log = self.baseFilename + ".1.gz"
+            os.remove(oldest_gz_log)
+            # Rename the remaining gzip'd logs...
+            arch_files = glob.iglob(base_file_pattern)
+            for archFile in arch_files:
+                fn = archFile.split('.')
+                # The gzip'd file "number" is in array element 2
+                newgz_file_num = int(fn[2]) - 1
+                newgz_file = fn[0] + "." + fn[1] + "." + str(newgz_file_num) + ".gz"
+                os.rename(archFile, newgz_file)
+        else:
+            # haven't hit the maximum backup file count yet, so simply construct the
+            # next gzip file name based on the existing base file name components
+            fn = (self.baseFilename + ".1.gz").split('.')
+
+        nextgz_file_name = fn[0] + "." + fn[1] + "." + str(nextgz_file_int) + ".gz"
+        dest = self.baseFilename + ".1"
+
+        # Start a thread to gzip the output file so as not to delay primary task(s)
+        done_flag = False
+        ex = futures.ThreadPoolExecutor(max_workers=2)
+        print('Starting GZIP thread for log file {dest}...')  # debug
```

Comment:
Did you mean to leave this debug output in?

Reply:
Not necessarily...I can take it out but it doesn't hurt.

Reply:
My thought was that -- especially if this moves up to `ECC.py` -- we don't need to see this output in regular runs.

If you have some debugging output that's useful to keep, it might be useful to put in a flag specific to this class (e.g., `self.debug`) that, if true, will emit additional debugging output.

Reply:
I see your point about its use as a generic handler...I'll remove it.